langcodes/__init__.py

```diff
@@ -250,31 +258,32 @@ def get(tag: str, normalize=True) -> 'LanguageData':
             else:
                 data[typ] = value
 
-        return LanguageData(**data)
+        result = Language.make(**data)
+        Language._PARSE_CACHE[tag, normalize] = result
+        return result
 
     def to_tag(self) -> str:
         """
-        Convert a LanguageData back to a standard language tag, as a string.
-        This is also the str() representation of a LanguageData object.
+        Convert a Language back to a standard language tag, as a string.
+        This is also the str() representation of a Language object.
 
-        >>> LanguageData(language='en', region='GB').to_tag()
+        >>> Language.make(language='en', region='GB').to_tag()
         'en-GB'
 
-        >>> LanguageData(language='yue', macrolanguage='zh', script='Hant',
-        ...              region='HK').to_tag()
+        >>> Language.make(language='yue', script='Hant', region='HK').to_tag()
         'yue-Hant-HK'
 
-        >>> LanguageData(script='Arab').to_tag()
+        >>> Language.make(script='Arab').to_tag()
         'und-Arab'
 
-        >>> str(LanguageData(region='IN'))
+        >>> str(Language.make(region='IN'))
         'und-IN'
         """
+        if self._str_tag:
+            return self._str_tag
```

Comment:
I was going to say "should this say `is None` for consistency", but I now wonder if all of this wants to use the lazy property decorator from our science code.

Maybe that would be an API change, though.